app/models/compute_resource.rb

```diff
@@ -244,7 +244,16 @@ def compute_profile_attributes_for(id)
 
   def vm_compute_attributes_for(uuid)
     vm = find_vm_by_uuid(uuid)
-    vm.attributes.reject{|k,v| k == :id }
+    vm_attrs = vm.attributes rescue {}
+    vm_attrs = vm_attrs.reject{|k,v| k == :id }
+
+    if vm.respond_to?(:volumes)
+      volumes = vm.volumes || []
+      vm_attrs[:volumes_attributes] = Hash[volumes.each_with_index.map { |volume, idx| [idx.to_s, volume.attributes] }]
+    end
+    vm_attrs
+  rescue ActiveRecord::RecordNotFound
```

Comment:
That's the unification of `find_vm_by_uuid(uuid)` exceptions. When the vm is not found `RecordNotFound` is raised.